server/src/internalClusterTest/java/org/elasticsearch/common/network/ThreadWatchdogIT.java

```diff
@@ -0,0 +1,163 @@
+            .put(ThreadWatchdog.NETWORK_THREAD_WATCHDOG_INTERVAL.getKey(), "100ms")
+            .put(ThreadWatchdog.NETWORK_THREAD_WATCHDOG_QUIET_TIME.getKey(), "0")
+            .build();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return CollectionUtils.appendToCopyNoNullElements(
+            super.nodePlugins(),
+            SlowRequestProcessingPlugin.class,
+            MockTransportService.TestPlugin.class
+        );
+    }
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false;
+    }
+
+    public static class SlowRequestProcessingPlugin extends Plugin implements ActionPlugin {
+
+        @Override
+        public Collection<RestHandler> getRestHandlers(
+            Settings settings,
+            NamedWriteableRegistry namedWriteableRegistry,
+            RestController restController,
+            ClusterSettings clusterSettings,
+            IndexScopedSettings indexScopedSettings,
+            SettingsFilter settingsFilter,
+            IndexNameExpressionResolver indexNameExpressionResolver,
+            Supplier<DiscoveryNodes> nodesInCluster,
+            Predicate<NodeFeature> clusterSupportsFeature
+        ) {
+            return List.of(new RestHandler() {
+                @Override
+                public List<Route> routes() {
+                    return List.of(Route.builder(RestRequest.Method.POST, "_slow").build());
+                }
+
+                @Override
+                public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) {
+                    blockAndWaitForWatchdogLogs();
+                    new RestToXContentListener<>(channel).onResponse((b, p) -> b.startObject().endObject());
+                }
+            });
+        }
+    }
+
+    private static void blockAndWaitForWatchdogLogs() {
+        final var threadName = Thread.currentThread().getName();
+        final var logsSeenLatch = new CountDownLatch(2);
+        final var warningSeen = new RunOnce(logsSeenLatch::countDown);
+        final var threadDumpSeen = new RunOnce(logsSeenLatch::countDown);
+        MockLog.assertThatLogger(() -> safeAwait(logsSeenLatch), ThreadWatchdog.class, new MockLog.LoggingExpectation() {
+            @Override
+            public void match(LogEvent event) {
+                final var formattedMessage = event.getMessage().getFormattedMessage();
+                if (formattedMessage.contains("the following threads are active but did not make progress since the last scan")
+                    && formattedMessage.contains(threadName)) {
+                    warningSeen.run();
+                }
+                if (formattedMessage.contains("hot threads dump due to active threads not making progress")) {
+                    threadDumpSeen.run();
+                }
+            }
+
+            @Override
+            public void assertMatched() {}
+        });
```

Comment:
This usage feels a bit abusive for the mock logger. But I guess it's ok for testing code.

Reply:
Seems fine to me - not sure how else we'd do this.